src/Tizen.System.Session/Session/Session.cs

```diff
@@ -0,0 +1,509 @@
+                {
+                    task.SetException(exception);
+                    return;
+                }
+
+                task.SetResult(true);
+            };
+
+            SessionError ret = Interop.Session.SubsessionSwitchUser(SessionUID, userName, _replyMap[taskID], (IntPtr)taskID);
+            CheckError(ret, "Interop failed to register a reply for switching a user");
+            return task.Task;
+        }
+
+        /// <summary>
+        /// Mark event as completed.
+        /// </summary>
+        /// <param name="subsessionEventArgs">Event argument of the event (obtained from said event)</param>
+        /// <remarks>
+        /// This method is assumed to be called from an event handler. You can only mark an event as completed
+        /// if you registered for in in the same process.
+        /// </remarks>
+        /// <exception cref="ArgumentException">Session UID of this object is invalid</exception>
+        /// <exception cref="IOException">Internal error</exception>
+        /// <exception cref="UnauthorizedAccessException">Not permitted</exception>
+        /// <exception cref="NotSupportedException">Not supported</exception>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public void SubsessionEventMarkAsDone(SubsessionEventArgs subsessionEventArgs)
+        {
+            SessionError ret = Interop.Session.SubsessionEventWaitDone(subsessionEventArgs.SessionInfo);
+            CheckError(ret, $"Interop failed to mark this client's event (of type {subsessionEventArgs.SessionInfo.eventType}) as finished");
+        }
+
+        private void OnAddUserWait(SubsessionEventInfoNative infoNative, IntPtr data)
+        {
+            _addUserWaitHandler?.Invoke(this, new AddUserEventArgs(infoNative));
+        }
+
+        private Interop.Session.SubsessionEventCallback _addUserWaitCB = null;
+
+        private event EventHandler<AddUserEventArgs> _addUserWaitHandler = null;
+
+        private readonly object _addUserWaitLock = new object();
+
+        /// <summary>
+        /// Event to be invoked when a new subession user is successfully added to this session.
+        /// </summary>
+        /// <exception cref="ArgumentException">Session UID of this object is invalid</exception>
+        /// <exception cref="OutOfMemoryException">Out of memory</exception>
+        /// <exception cref="IOException">Internal error</exception>
+        /// <exception cref="UnauthorizedAccessException">Not permitted</exception>
+        /// <exception cref="NotSupportedException">Not supported</exception>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public event EventHandler<AddUserEventArgs> AddUserWait
+        {
+            add
+            {
+                lock (_addUserWaitLock)
+                {
+                    if (_addUserWaitHandler == null)
+                        RegisterCallbackForEvent(SessionEventType.AddUserWait, _addUserWaitCB, OnAddUserWait);
```

Comment:
_addUserWaitCB is null
nowhere set content of _addUserWaitCB
I can't find it

Reply:
This is set in #495 in `RegisterCallbackForEvent` function:

    eventCallback = new Interop.Session.SubsessionEventCallback(delegateToSet);